### PR TITLE
[clang-tidy] NO.48-50 enable `string-integer-assignment`,`misplaced-widening-cast`,`infinite-loop`

### PR DESCRIPTION
**Pull request by enkilee**
### PR types

Others
### PR changes

Others
### Description

* https://github.com/PaddlePaddle/Paddle/issues/54073
fix 48-50

### CODE REVIEW

File: paddle/fluid/framework/ir/graph_helper.cc [hunk 1]
```diff
@@ -297,21 +297,19 @@ std::vector<ir::Node *> TopologyDfsSortOperations(const Graph &graph) {
 
   // traverse the graph
   int num_ops = static_cast<int>(op_queue.size());
-  while (num_ops) {
```
Comment (zhangbo9674) [inline paddle/fluid/framework/ir/graph_helper.cc:300]: 删除外层循环原因是什么？

Comment (enkilee) [reply paddle/fluid/framework/ir/graph_helper.cc:300]: /paddle/paddle/fluid/framework/ir/graph_helper.cc:299:3: error: this loop is infinite; none of its condition variables (num_ops) are updated in the loop body [bugprone-infinite-loop,-warnings-as-errors]

Comment (luotao1) [reply paddle/fluid/framework/ir/graph_helper.cc:300]: @enkilee 敏师傅，别忘了回复下哈

**Comment by luotao1** [comment]
敏师傅可以直接更新 tracking issue，把 PR号都填上去

**Comment by enkilee** [comment]
这个循环，num_ops--在for里面，while条件无限循环

File: paddle/fluid/framework/ir/graph_helper.cc (hunk 1 above)
Comment (enkilee) [reply paddle/fluid/framework/ir/graph_helper.cc:300]: 这个循环，num_ops--在for里面，while条件无限循环

**Review by zhangbo9674 — APPROVED** [review]

**Pull request merged**